app-office/texmacs/texmacs-1.99.12.recipe

```diff
@@ -0,0 +1,102 @@
+SUMMARY="A scientific editing platform designed to create beautiful technical documents"
+DESCRIPTION="TeXmacs provides a unified and user friendly framework for editing \
+structured documents with different types of content: text, mathematics, graphics, \
+interactive content, slides, etc.\
+Its rendering engine uses high-quality typesetting algorithms so as to produce \
+professionally looking documents, which can either be printed out or presented from \
+a laptop."
+HOMEPAGE="http://www.texmacs.org/"
+COPYRIGHT="Free Software Foundation"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/mgubi/texmacs/archive/texmacs-$portVersion-haiku-qt5.tar.gz"
+CHECKSUM_SHA256="607fa4db4676b78780f7af086feec2e2332dc18a371d79fa50d640de6db10b43"
+SOURCE_DIR="texmacs-texmacs-$portVersion-haiku-qt5/src"
+PATCHES="texmacs-$portVersion.patchset"
+ADDITIONAL_FILES="texmacs.rdef.in"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	texmacs$secondaryArchSuffix = $portVersion
+	app:TeXmacs
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libQt5Concurrent$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libguile$secondaryArchSuffix <= 19
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libQt5Concurrent$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libguile$secondaryArchSuffix <= 19
+	devel:libfreetype$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	cmd:guile_config$secondaryArchSuffix <= 1.9
```

Comment:
It should be 1.8.8, as guile recipe says:
```cmd:guile_config$commandSuffix = $portVersion```

Reply:
I understand but (see below) I want to leave the user the freedom to choose different but compatible versions of guile. No need to overcontraint. TeXmacs will work just fine with any guile < 2.0.

Reply:
Not needed, devel:libguile$secondaryArchsuffix provides cmd:guile_config$commandSuffix

Reply:
I was thinking that the idea is to have all the requirements explicit. I need here both the libraries and the guile_config script which are two independent resolvable. My understanding of the recommendations in the docs is that I need to explicitly put all the requirement, whether or not they are currently packaged together.

Reply:
cmd:guile_config is used to find the libraries headers needed to build TeXmacs, don't think they are required to run TeXmacs? (same for cmd:guile)?

Reply:
TeXmacs configuration script requires guile to be present in the path, otherwise abort with error. it also requires guile_config for the same reason. that is why I put them in build requirements but not in the runtime requirements.